### PR TITLE
integrate previous work

### PR DESCRIPTION
**Pull request by s-pace**
1 st preview

### CODE REVIEW

**Comment by s-pace** [comment]
algolia/docsearch-scraper#359

**Comment by s-pace** [comment]
👍 👍 @codeharmonics  Thanks.

Waiting for @maxiloc final review and I merge it

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb [hunk 1]
```diff
@@ -152,113 +223,130 @@ then for each record the default value will be set
...
 
-The `js_wait` parameter lets you change the default waiting time to render the
-webpage with the Selenium proxy.
+Default is `false`
 
-This parameter is optional and is set to `0`s by default.
+### `js_wait` _Optional_
 
-### `use_anchors`
+The `js_wait` parameter lets you change the default waiting time (in seconds) to render the
```
Comment (maxiloc) [inline docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: Not only for this one but I would mention dependency. In this case this option only makes sense when you have js_render: true

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb [hunk 2]
```diff
@@ -152,113 +223,130 @@ then for each record the default value will be set
...
-### `nb_hits`
+This attribute is used for purposed monitoring. We keep a track of its evolution in order to detect main changes.
+Default is `0`.
 
-The number of object that should be indexed. Only used by the [`checker`](#checker).
+### `custom_settings` _Optional_
 
-Default is `0`.
+This object is any custom Algolia setting you would like to pass to the index
+settings. Take a [look under the hood of algolia](https://www.algolia.com/doc/).
```
Comment (maxiloc) [inline docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: I would link to the settings parameters page

File: docs/source/documentation/1-docsearch/3-recommendations.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,92 @@
+---
+title: Recommendations
+---
+
```
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/1-docsearch/3-recommendations.html.md.erb]: It think ti spoil the leaning by intending to be too straight to the point. 

The `do this, do` POV is too related to the assumed use case. It is more relevant to give the big picture and further details even if this POV will be nice in a FAQ

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb [hunk 4]
```diff
@@ -152,113 +223,130 @@ then for each record the default value will be set
...
+### `js_render` _Optional_
 
 The HTML code that we crawl is sometimes generated using Javascript. In those
 cases, the `js_render` option must be set to `true`. It will enable our
 internal proxy (Selenium) to render pages before crawling them.
 
-This parameter is optional and is set to `false` by default.
-
-### `js_wait`
+We highly recommend avoiding client-side rendering. It mainly decreases the performance of your website.
```
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: Should be rephrase but we also need to let people know that it is more costly on our side. Last but not least it often introduces bigger issue. The simpler, the better

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb [hunk 5]
```diff
@@ -53,67 +53,139 @@ A configuration file looks like:
...
-with it.
+## DocSearch options
 
-*This name must be equal to the configuration file name*
+### `index_name` _Mandatory_
 
-### `start_urls` ***Mandatory***
+Name of the Algolia index where all the data will be pushed. If the `PREFIX` environment variable is defined, it will be prefixed with it.
+
+**This name must be equal to the configuration file name**
```
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: We must have this equality `index_name`:`filename`.json. It is better to clearly write it down

File: docs/source/documentation/1-docsearch/3-recommendations.html.md.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxiloc) [reply docs/source/documentation/1-docsearch/3-recommendations.html.md.erb]: imo people would understand the same from both

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb [hunk 6]
```diff
@@ -53,67 +53,139 @@ A configuration file looks like:
...
+
+We mostly attribute it on our own regarding plenty of underlying factors. The `apiKey` provided is highly correlated to the `index_name`. Thus if you want to **change the name**, please **submit a new configration**, we will generate a new key accordingly.
```
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: 👍

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (maxiloc) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: That's our issue, not theirs 😄. For me it's more relevant when they use it on their side

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb [hunk 7]
```diff
@@ -152,113 +223,130 @@ then for each record the default value will be set
...
+### `selectors_exclude` _Optional_
 
-You can override the default `strip_chars` per level
+By default, the `selectors` search is applied page-wide. If there are some parts
+of the page that you do not want to include (e.g. a table of content, a sidebar or a footer),
+you can add them to the `selectors_exclude` key.
+
+### Sitemap crawling _Optional_
 
+Our crawler offers you to crawl a site by discovering the URLs using Sitemaps. Thus, you will need to define the direct url(s) to your sitemap XML file, `sitemap_urls` , and then establish regex(s), `sitemap_urls_regexs` , which will match the URLs to crawl. Otherwise it will use the `start_urls` pattern in order to match the URLs available within the site map.
```
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: I will try to rephrase it

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb (hunk 5 above)
[the remark this answers is not in the document]
Comment (maxiloc) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: yes for us, but not when they run it on their side

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: Given the current state, it is only theirs for now. We can change it later on. Most of the time, when they use client-side rendering, it is not in a proper way (js_wait ... )

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb (hunk 5 above)
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: Yes but it is the biggest use case and this is why it isn't forced by our tools

File: docs/source/documentation/1-docsearch/3-recommendations.html.md.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/1-docsearch/3-recommendations.html.md.erb]: I will change it if needed but I think it is downgrading the doc

File: docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb [hunk 8]
```diff
@@ -53,67 +53,139 @@ A configuration file looks like:
...
+will not consider their content. Likewise, if a link within a crawled webpage
+targets such pages, the crawler will not follow the link.
+You can use a regular expression as well as plain urls.
 
-### Global selectors
+Note: It is sometimes needed to add `http://www.example.com/index.html` pages to
+the `stop_urls` list if you set `http://www.example.com` as a `start_urls`, to
+avoid duplicated content.
+
+### `selectors` _Mandatory_
```
[the remark this answers is not in the document]
Comment (s-pace) [reply docs/source/documentation/2-docsearch-scraper/2-config-options.html.md.erb]: In order to make them more visible, I will change the style of the mandatory elements if needed. I would rather the attributes to follow a standard structure.

**Pull request merged**